From c191ff37a6998471dd37d6f5f383236a2e6f492c Mon Sep 17 00:00:00 2001
From: Niko Tyni <ntyni@debian.org>
Date: Tue, 10 Jan 2012 20:28:48 +0200
Subject: [PATCH 1/3] Make the embedded File::Which::which() work in BEGIN
 blocks

The File::Which::which() subroutine uses a lexical variable @path_ext,
which used to be uninitialized when the subroutine was called
from BEGIN blocks.

This included the 'use constant' lines, making both HAVEZIP and HAVEUNZIP
constants unconditionally false and disabling 'testzip' functionality
in for instance t/02_main.t:

 ok 132 # skip No 'unzip' program to test against
---
 t/common.pl |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/t/common.pl b/t/common.pl
index 808d6b8..1560fd2 100644
--- a/t/common.pl
+++ b/t/common.pl
@@ -7,6 +7,7 @@ use constant OUTPUTZIP => 'testout.zip';
 
 # Do we have the 'zip' and 'unzip' programs?
 # Embed a copy of the module, rather than adding a dependency
+BEGIN {
 SCOPE: {
 	package File::Which;
 
@@ -103,6 +104,7 @@ SCOPE: {
 	    }
 	}
 }
+}
 use constant HAVEZIP   => !! File::Which::which('zip');
 use constant HAVEUNZIP => !! File::Which::which('unzip');
 
-- 
1.7.8.2

